File: LeetCode Questions + Tutorials/EngineeringDigest/Leetcode1207.java
import java.util.*;

class Leetcode1207 {
    public static void main(String[] args) {
        Scanner input = new Scanner(System.in);
        System.out.print("Enter the size of the array: ");
        int size = input.nextInt();
        int[] arr = new int[size];
        for(int i = 0; i < size; i++) {
            System.out.printf("Enter %d number element of the array: ", i + 1);
            arr[i] = input.nextInt();
        }
        input.close();
        System.out.println(uniqueOccurrences(arr));
    }

    private static boolean uniqueOccurrences(int[] arr) {
        HashMap<Integer, Integer> map = new HashMap();
        Set<Integer> set = new HashSet<>();
        for(int i : arr) {
            map.put(i, map.getOrDefault(i, 0) + 1);
        }
        for(Map.Entry<Integer, Integer> entry : map.entrySet()) {
            if(set.contains(entry.getValue())) {
                return false;
            }else {
                set.add(entry.getValue());
            }
        }
        return true;
    }
}

File: LeetCode Questions + Tutorials/EngineeringDigest/Leetcode1207.js
function uniqueOccurrences(arr) {
    var map = new Map();
    var set = new Set();
    for (var _i = 0, arr_1 = arr; _i < arr_1.length; _i++) {
        var i = arr_1[_i];
        map.set(i, (map.get(i) || 0) + 1);
    }
    for (var _a = 0, _b = map.entries(); _a < _b.length; _a++) {
        var _c = _b[_a], key = _c[0], value = _c[1];
        if (set.has(value)) {
            return false;
        }
        else {
            set.add(value);
        }
    }
    return true;
}
;
var arr = [1, 2, 2, 1, 1, 3];
console.log(uniqueOccurrences(arr));
